File: shoppingsystem/server/models/Banner.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Banner = sequelize.define('Banner', {
  id: { type: DataTypes.BIGINT, primaryKey: true, autoIncrement: true },
  image_url: { type: DataTypes.STRING(500), allowNull: false },
  link_type: { type: DataTypes.ENUM('PRODUCT', 'CATEGORY', 'NONE'), defaultValue: 'NONE' },
  link_value: { type: DataTypes.BIGINT, defaultValue: 0 },
  sort_order: { type: DataTypes.INTEGER, defaultValue: 0 }
}, {
  tableName: 'banners',
  timestamps: false
});

module.exports = Banner;

File: shoppingsystem/app/src/main/java/com/example/shoppingsystem/data/local/entity/Banner.java
package com.example.shoppingsystem.data.local.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "banners")
public class Banner {

    @PrimaryKey(autoGenerate = true)
    private long id;

    @ColumnInfo(name = "image_url")
    private String imageUrl;

    @ColumnInfo(name = "link_type")
    private String linkType; // "PRODUCT", "CATEGORY", or "NONE"

    @ColumnInfo(name = "link_value")
    private long linkValue; // productId or categoryId

    @ColumnInfo(name = "sort_order")
    private int sortOrder;

    public Banner() {}

    public Banner(String imageUrl, String linkType, long linkValue, int sortOrder) {
        this.imageUrl = imageUrl;
        this.linkType = linkType;
        this.linkValue = linkValue;
        this.sortOrder = sortOrder;
    }

    // Getters and Setters

    public long getId() { return id; }
    public void setId(long id) { this.id = id; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }

    public String getLinkType() { return linkType; }
    public void setLinkType(String linkType) { this.linkType = linkType; }

    public long getLinkValue() { return linkValue; }
    public void setLinkValue(long linkValue) { this.linkValue = linkValue; }

    public int getSortOrder() { return sortOrder; }
    public void setSortOrder(int sortOrder) { this.sortOrder = sortOrder; }
}
